src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/GestureManager.shared.cs

```diff
@@ -202,13 +204,52 @@ internal void OnTapped(TouchEffect sender)
 			if (!sender.CanExecute || (sender.LongPressCommand != null && sender.InteractionStatus == TouchInteractionStatus.Completed))
 				return;
 
+			if (Device.RuntimePlatform == Device.Android)
+				HandleCollectionViewSelection(sender);
+
 			if (sender.Element is IButtonController button)
 				button.SendClicked();
 
 			sender.Command?.Execute(sender.CommandParameter);
 			sender.RaiseCompleted();
 		}
 
+		void HandleCollectionViewSelection(TouchEffect sender)
+		{
+			Element element = sender.Element;
+			while (element.Parent != null)
+			{
+				if (!(element.Parent is CollectionView collectionView))
+				{
+					element = element.Parent;
+					continue;
+				}
+
+				var item = element.BindingContext ?? element;
+
+				switch (collectionView.SelectionMode)
+				{
+					case SelectionMode.Single:
+						var selectedItem = collectionView.SelectedItem;
+						collectionView.SelectedItem = selectedItem == item
+							? null
+							: item;
+						break;
+					case SelectionMode.Multiple:
+						var selectedItems = collectionView.SelectedItems?.ToList() ?? new List<object>();
```

Comment:
I not sure if this makes sense here because I can't imagine a scenario where we will have a lot of selectedItems, but if we will integrate from items to remove then (line 241) could be better to use a [`HashSet` ](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.tohashset?view=net-5.0)instead of a `List`

Reply:
In this case, I think, XF should provide HashSet as well (now it is List in SelectedItems).

The current complexity is:
1) Create a new list of elements O(n)
2) Check if the SINGLE element is inside the list O(n)
3) Remove the SINGLE element O(n)

Additional memory N (one list)
it's linear complexity O(n)

If we go ahead with HashSet it will be
1) Add all elements to set O(n)
2) Find if the element is inside HashSet O(1)
3) Remove the element from the set O(1)
4) Invoke UpdateSelectedItems (which requires List) it means creating List O(n)

Additional memory 2*N (HashSet + List)
But it's still linear complexity